compiler/src/iree/compiler/Dialect/HAL/Target/CUDA/CUDATarget.cpp

```diff
@@ -409,8 +410,12 @@ class CUDATargetBackend final : public TargetBackend {
       workgroupLocalMemories.push_back(workgroupLocalMemory);
     }
 
-    SmallVector<std::string> entryPointNames;
+    FlatbufferBuilder builder;
+    iree_CUDAExecutableDef_start_as_root(builder);
+
+    SmallVector<std::string, 8> entryPointNames;
```

Comment:
nit: Isn't this a lot of stack space? IIRC, strings are ~32 bytes large, so here alone we have >256B of stack space consumed. Consider using the default size.

Reply:
I was copying what `VulkanSPIRVTarget.cpp` has: https://github.com/openxla/iree/blob/5d15caf4bc2eefbc517461a2d78ae9b21008f40c/compiler/src/iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp#L164-L165

I'd expect the actual number of entry points to be 1-100 depending on how linking / archives are handled, see also https://github.com/openxla/iree/issues/7824)

Reply:
`StringRef` (ptr + len) is much smaller than `std::string`. In any case, I don't have a strong opinion on this, feel free to keep this as-is if you prefer.